daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1072,6 +1072,115 @@ private[lf] object SBuiltin {
     }
   }
 
+  // Similar to SBUFetch but doesn't perform any checks on the template id, and is never "by key".
+  final case class SBUPreFetchInterface(ifaceId: TypeConName) extends OnLedgerBuiltin(1) {
+    override protected def execute(
+        args: util.ArrayList[SValue],
+        machine: Machine,
+        onLedger: OnLedger,
+    ): Unit = {
+      val coid = getSContractId(args, 0)
+      onLedger.cachedContracts.get(coid) match {
+        case Some(cached) => {
+          machine.returnValue = cached.value
+        }
+        case None => {
+          throw SpeedyHungry(
+            SResultNeedContract(
+              coid,
+              ifaceId, // not actually used, maybe this param should be dropped from SResultNeedContract
```

Comment:
hm this is somewhat interesting. We considered at some point to pass this along to ledger implementations so they could use this to make lookups on their side more efficient. I think interfaces don’t fundamentally break this but at least simple table partitioning by template gets more difficult. Anyway, we never actually went down that route so I don’t think this is blocking by any means, just recording my thoughts.

Reply:
Maybe it will be worthwhile to ask Simon, his option.